File: 2023/2023W01/preppindataW01.R
library(dplyr)
library(readr)
library(stringr)

Sys.setlocale("LC_ALL","English")

txndata <- read_csv("C:/Data/PreppinData/PD 2023 Wk 1 Input.csv", 
                  col_types = cols(`Transaction Date` = col_date(format = "%d/%m/%Y %H:%M:%S"))) %>%
  mutate('Bank' = str_extract(`Transaction Code`,'^([A-Z]+)'),
         'Online or In-Person' = if_else(`Online or In-Person`==1, 'In-Person', 'Online'),
         'Transaction Date' = strftime(`Transaction Date`, '%A'))

out1 <- txndata %>%
  group_by(`Bank`) %>%
  summarise('Value'=sum(`Value`), .groups='drop')

out2 <- txndata %>%
  group_by(`Bank`, `Online or In-Person`, `Transaction Date`) %>%
  summarise('Value'=sum(`Value`), .groups='drop')

out3 <- txndata %>%
  group_by(`Bank`, `Customer Code`) %>%
  summarise('Value'=sum(`Value`), .groups='drop')
  
View(out1)
View(out2)
View(out3)

File: 2023/2023W01/PreppindataW01.py
import pandas as pd
import re

final = pd.read_csv(r'C:/Data/PreppinData/PD 2023 Wk 1 Input.csv',
                    parse_dates=['Transaction Date'], date_parser=lambda x: pd.to_datetime(x, format='%d/%m/%Y %H:%M:%S'))

final['Bank'] = final['Transaction Code'].apply(lambda x: re.search('^([A-Z]+)',x).group(1))
final['Online or In-Person'] = final['Online or In-Person'].apply(lambda x: 'Online' if x==1 else 'In-Person')  
final['Transaction Date'] = final['Transaction Date'].dt.day_name()

output1 = final.groupby(['Bank'], as_index=False).agg({'Value':'sum'})
output2 = final.groupby(['Bank', 'Online or In-Person', 'Transaction Date'], as_index=False).agg({'Value':'sum'})
output3 = final.groupby(['Bank', 'Customer Code'], as_index=False).agg({'Value':'sum'})
